src/features/core/metrics/errorReporting/providers/Bugsnag.js

```diff
@@ -0,0 +1,240 @@
+/* eslint-disable no-underscore-dangle */
+import ScriptLoader from '../../../../../utils/ScriptLoader';
+import { configToIntNames } from '../../../../../utils/config_to_integration_names';
+import {
+  ERROR_REPORTING_SERVICE_GLOBAL_KEY_NAME,
+  MAX_WAIT_FOR_INTEGRATION_LOAD,
+} from '../../../../../utils/constants';
+import { get } from '../../../../../utils/utils';
+
+// Using the Bugsnag integration version to avoid version issues
+const BUGSNAG_CDN_URL = 'https://d2wy8f7a9ursnm.cloudfront.net/v6/bugsnag.min.js';
+const BUGSNAG_VALID_MAJOR_VERSION = '6';
+const ERROR_REPORT_PROVIDER_NAME_BUGSNAG = 'rs-bugsnag';
+const BUGSNAG_LIB_INSTANCE_GLOBAL_KEY_NAME = 'bugsnag'; // For version 7 this is 'Bugsnag'
+const GLOBAL_LIBRARY_OBJECT_NAMES = [
+  ERROR_REPORT_PROVIDER_NAME_BUGSNAG,
+  'Bugsnag',
+  BUGSNAG_LIB_INSTANCE_GLOBAL_KEY_NAME,
+  ERROR_REPORTING_SERVICE_GLOBAL_KEY_NAME,
+];
+
+// This API key token is parsed in the CI pipeline
+const API_KEY = '__RS_BUGSNAG_API_KEY__';
+
+// Errors only from Below SDKs are allowed to reach Bugsnag
+const SDK_FILE_NAMES = [
+  'rudder-analytics.min.js',
+  'rudder-analytics-staging.min.js',
+  'rudder-analytics.js',
+  ...Object.keys(configToIntNames).map((intgName) => `${configToIntNames[intgName]}.min.js`),
+  ...Object.keys(configToIntNames).map(
+    (intgName) => `${configToIntNames[intgName]}-staging.min.js`,
+  ),
+  ...Object.keys(configToIntNames).map((intgName) => `${configToIntNames[intgName]}.js`),
+];
+const BUGSNAG_ERROR_FILE_ORIGIN_KEY = 'stacktrace.0.file';
+
+const getReleaseStage = () => {
+  const host = window.location.hostname;
+  const devHosts = ['www.rs-unit-test-host.com', 'localhost', '127.0.0.1', '[::1]'];
+
+  return host && devHosts.includes(host) ? 'development' : '__RS_BUGSNAG_RELEASE_STAGE__';
+};
+
+const isValidVersion = (globalLibInstance) => {
+  // For version 7
+  let version =
+    globalLibInstance &&
+    globalLibInstance._client &&
+    globalLibInstance._client._notifier &&
+    globalLibInstance._client._notifier.version;
```

Comment:
Would this work in v7 as per the comment above the 'global lib name' is different that what was used?

Reply:
that is not the case v7 global objet is named Bugsnag. This is part of the GLOBAL_LIBRARY_OBJECT_NAMES array.

SDK will not attempt to load if any of the items in the array exists. (L102) 
https://github.com/rudderlabs/rudder-sdk-js/pull/856/files#diff-660103bc747839cc9cbabf3b82633c13ea449e2f61e6e2d65595a5436c1e2847R102

If any version of bugsnag that is not v6.x exists we are not loading it